Mathlib/Topology/Compactness/Compact.lean

```diff
@@ -303,6 +303,15 @@ theorem IsCompact.nonempty_iInter_of_directed_nonempty_compact_closed {ι : Type
   exact (htn j).mono (subset_inter hji₀ hji)
 #align is_compact.nonempty_Inter_of_directed_nonempty_compact_closed IsCompact.nonempty_iInter_of_directed_nonempty_compact_closed
 
+/-- Cantor's intersection theorem for `sInter`:
+the intersection of a directed family of nonempty compact closed sets is nonempty. -/
+theorem IsCompact.nonempty_sInter_of_directed_nonempty_compact_closed
```

Comment:
```suggestion
theorem IsCompact.nonempty_sInter_of_directed_nonempty_isCompact_isClosed
```
We try to have theorem names that follow as closely as possible the Lean names. The other `IsCompact.nonempty_iInter_of_directed_nonempty_compact_closed` is also wrong, by the way -- could you correct it, and add a deprecated alias for the old name? (If you don't know how to do that, tell me)

Reply:
Sure thing, I've corrected the name of the newly added theorem. 

I've updated the naming of the other two cantor intersection theorems and added the alias. Shall I do this as part of this PR or better as a separate PR? I've done them here but let me know and I can revert this and do a separate PR for the name change.

The mathlib style guide doesn't tell me the preferred place for the deprecated alias. Sometimes they are the end of the files and sometimes they are immediately after the associated theorem. I've put it after the theorem with a single space between the two since there is already the #align there. Otherwise at the end of the file seems more tidy to me. Give me guidance and then I can change is (and also add a note to the style guide if there is a maintainer consensus.)